File: 0024-swap-nodes-in-pairs/0024-swap-nodes-in-pairs.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
        ListNode* res = new ListNode(0);
        res->next = head;
        ListNode* curr=res;
        while(curr->next && curr->next->next ){
            ListNode* first=curr->next;
            ListNode* second=curr->next->next;
            first->next=second->next;
            curr->next=second;
            second->next=first;
            curr=first;
        }
        return res->next;
    }
};

File: 0024-swap-nodes-in-pairs/0024-swap-nodes-in-pairs.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def swapPairs(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if head==None or head.next==None:
            return head
        curr,prev = head,ListNode(0)
        res=prev
        while curr and curr.next:
            nxt=curr.next
            fast=nxt.next
            nxt.next=curr
            curr.next=fast
            prev.next=nxt
            prev=curr
            curr=fast
            
        return res.next